### DISCUSSION TITLE
Easily connect chatflows via ID

### DISCUSSION THREAD

**Discussion by cjaysd** [FlowiseAI/Flowise#776] [Ideas]
I have not seen this capability in the documentation and I think this could create rather powerful flows if prompt templates could intake the output of one flow as system instructions, actual prompt, or contextual documents. 

In general : Has anyone seen a capability that allows you to feed documents processed by Conversational Retrieval QA as an input to another agent? It seemed like the "Retriever Tool" might be utilized for something like this, but it is not functioning for me.

**Comment by HenryHengZJ** [comment]
one way to achieve is to use retrievertool, where the agent decide when to use the context fetched from vector store to build response.
another way (not yet pushed to new release), is use prompt chaining. This [PR](https://github.com/FlowiseAI/Flowise/pull/775) allow you to use output from a chain to another prompt, and the documents fetched from vector store as `{context}` in the prompt, and the past chat history as `{chat_history}`

<img width="1261" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/26460777/cbd55ade-8512-4369-9139-f3c438dbae68">

**Reply by jcalebsmith** [reply]
Oh wow, that PR is *exactly* what I've been looking for!
